app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -30,9 +30,63 @@ interface FormattedLocation {
   weekdayHours: string;
 }
 
+interface RequestOptions {
+  /**
+   * Whether to send the request as a JSON request. Defaults to true.
+   */
+  json?: boolean;
+
+  /**
+   * Whether to include CSRF token in the request. Defaults to true.
+   */
+  csrf?: boolean;
+
+  /**
+   * Optional. HTTP verb used. Defaults to GET.
+   */
+  method?: string;
+}
+
+const DEFAULT_FETCH_OPTIONS = { csrf: true, json: true };
+
+const getCSRFToken = () =>
+  document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
+
+const request = async (
+  url: string,
+  body: BodyInit | object,
+  options: Partial<RequestOptions> = {},
+) => {
+  const headers: HeadersInit = {};
+  const mergedOptions: Partial<RequestOptions> = {
+    ...DEFAULT_FETCH_OPTIONS,
+    ...options,
+  };
+
+  if (mergedOptions.csrf) {
+    const csrf = getCSRFToken();
+    if (csrf) {
+      headers['X-CSRF-Token'] = csrf;
+    }
+  }
+
+  if (mergedOptions.json) {
+    headers['Content-Type'] = 'application/json';
+    body = JSON.stringify(body);
+  }
+
+  const response = await window.fetch(url, {
+    method: mergedOptions.method,
+    headers,
+    body: body as BodyInit,
+  });
+
+  return mergedOptions.json ? response.json() : response.text();
+};
+
 export const LOCATIONS_URL = '/verify/in_person/usps_locations';
 
-const getResponse = () => window.fetch(LOCATIONS_URL).then((res) => res.json());
+const getUspsLocations = () => request(LOCATIONS_URL, {}, { method: 'post' });
```

Comment:
To consider for follow-up if we refactor a new package: Maybe `body` should be an option as well, so that it's a little easier to leave empty like we've done here? There'd be some precedent from `method`, which is also passed as an option to `fetch`, so `RequestOptions` could be a superset of the default fetch options.

```ts
interface RequestOptions extends RequestInit {
  /**
   * Whether to send the request as a JSON request. Defaults to true.
   */
  json?: boolean;

  /**
   * Whether to include CSRF token in the request. Defaults to true.
   */
  csrf?: boolean;
}

async function request(url: string, options: RequestOptions = {}) {
  const { csrf = true, json = true, ...fetchOptions } = options;
  let { body, headers } = fetchOptions;
  headers = new Headers(headers);

  if (csrf) {
    const csrfToken = getCSRFToken();
    if (csrfToken) {
      headers.set('X-CSRF-Token', csrfToken);
    }
  }

  if (json) {
    headers.set('Content-Type', 'application/json');
    body = JSON.stringify(body);
  }

  const response = await fetch(url, { ...fetchOptions, headers, body });
  return json ? response.json() : response.text();
}
```